tools/check

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
```

Comment:
this looks overly complicated. why do we need/want this?

Reply:
See commit message.

Reply:
Or what do you mean? Any suggestion for how to do it in a simpler way? (As mentioned in commit message and comment in code, long term we want Meson targets. But that will probably take a while even if we do it ourselves.)

Reply:
FYI there is a shell version here: https://github.com/martin-ejdestig/template-dbus-service/blob/check_sh/tools/check

Not that much difference in line count (if one disregards space and imports etc.).

There are some downsides with it compared to Python version though (stops after first error, does not parallelize clang-format, relies on `run-clang-tidy` that is not installed in a "standard" location by most Linux distros, grep:s through clang-tidy output to check if there was an error instead of relying on exit code... I think that was all, do not remember).

Reply:
to not hold you on this: I would prefer this solution to be as simple as possible, but if you have made a research and couldn't find anything better, let's go with this one...

Reply:
I do not know if you prefer the Python version or Shell version.

Pushed a fixup commit with the shell version and added an update-alternatives in `tools/ci/Dockerfile` to find `run-clang-tidy`. It is ~27 lines of code compared to ~51 for the Python version (disregarding imports and double space lines in Python version and comments in both).